src/kbmod/regionsearch/region_search.py

```diff
@@ -0,0 +1,261 @@
+from astropy.time import Time
+from astropy.units import Quantity
+
+
+class RowSet(object):
+    """Container for region_search rows.
+
+    A region search row contains the columns used in the region search. Not all columns are needed at all times and the access pattern is unclear. Hence the RowSet, which abstracts the collection from the
+    details of the row. The columns of the collection are as follows:
+
+    - ``pointing``: Right ascension abd declination of the observation as a SkyCoord.
+    - ``time``: Time of the observation.
+    - ``fov``: The field of view of the observation in degrees.
+    - ``distance_far``: The farthest distance from the barycenter in which the observation appears with the index.
+    - ``distance_near``: The nearest distance from the barycenter in which the observation appears with the index.
+    - ``observatory_location``: The location of the observatory in geocentric coordinates. Together with the time this locates the origin of the observation.
+    - ``data_source``: The data source of the observation. This is a string that identifies the data source and may be None
+    - ``observation_id``: A unique identifier for the observation. Together with data_source this should provide enough information to trace back to the observation source.
+    - ``index``: Index of the observation.
+
+    All rows occur uniquely in the collection. The columns pointing, time, and fov are required. The columns index and distance_interval may either both be None or both not None.
+    The columns observatory_location, data_source and observation_id may be None.
+    If observatory_location is None then the observation is assumed to be geocentric.
+
+    The table is initialized in the __init__ or by calling the add method one or more times. The get_table method returns the data as an astropy.table and freezes the RowSet so that no more rows can be added.
+    Completely empty columns are left in the table with all None entries in a Column instance. This is done so that indexing columns and rows always returns data (even if just None).
+
+    """
+
+    names_type1 = ["pointing", "time", "fov", "distance_near", "distance_far", "observatory_location"]
+    names_type2 = ["data_source", "observation_id", "index"]
+    names = names_type1 + names_type2
+    dtype = [SkyCoord, Time, Quantity, Quantity, Quantity, EarthLocation, "str", "uint32", "uint32"]
+    units = [None, None, u.deg, u.au, u.au, None, None, None, None]
+
+    def __init__(self):
+        """Initializes the RowSet object."""
+        self.table = None
+        self.pointing = None
+        self.time = None
+        self.fov = None
+        self.distance_far = None
+        self.distance_near = None
+        self.observatory_location = None
+        self.data_source = None
+        self.observation_id = None
+        self.index = None
+        self.rowcnt = 0
+        self.dtype = RowSet.dtype.copy()
+
+    def add(self, **kwargs):
```

Comment:
We can revisit should RowSet resurface. There are reasons having to do with the edge cases of how astropy.table seems to handle Columns, mixin types (e.g. SkyCoord) and what happens with rows are first added and removed. I think a better approach is to avoid row insertions and removals.